### PR TITLE
Match SummonEnemy & UseIdleAnimation

### PR DESCRIPTION
**Pull request by Technoman589**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.02%) on `us`.

⚠️ This PR introduces 2 warnings:
- src/197F40.c:1963: warning: implicit declaration of function `create_actor'
- src/197F40.c:1969: warning: assignment makes pointer from integer without a cast
- src/197F40.c:2415: warning: implicit declaration of function `remove_player_buffs'
- src/197F40.c:2471: warning: implicit declaration of function `reset_all_actor_sounds'
- src/197F40.c:2483: warning: initialization makes pointer from integer without a cast
- src/197F40.c:2487: warning: comparison between pointer and integer
- src/197F40.c:2488: warning: assignment makes pointer from integer without a cast
- src/197F40.c:2492: warning: passing arg 1 of `get_actor' makes integer from pointer without a cast
- src/197F40.c:2494: warning: implicit declaration of function `is_actor_hp_bar_visible'
- src/197F40.c:2495: warning: implicit declaration of function `load_tattle_flags'
- src/197F40.c:2510: warning: implicit declaration of function `func_80251474'

File: src/197F40.c [hunk 1]
```diff
@@ -1942,7 +1942,95 @@ ApiStatus func_8026DA94(Evt* script, s32 isInitialCall) {
...
+        script->functionTemp[0] = 0;
+    }
+    
+    switch (script->functionTemp[0]) {
+        case 0:
+            script->functionTemp[1] = create_actor(evt_get_variable(script, *args++));
+            script->functionTemp[2] = evt_get_variable(script, *args++);
+            script->functionTemp[0] = 1;
+            break;
+
```
Comment (ethteck) [inline src/197F40.c:1967]: You can remove the empty lines between cases

File: src/197F40.c [hunk 2]
```diff
@@ -1942,7 +1942,95 @@ ApiStatus func_8026DA94(Evt* script, s32 isInitialCall) {
...
+                    for (i = 0; i < numEnemies - 1; i++) {
+                        for (j = i + 1; j < numEnemies; j++) {
+                            enemyID1 = enemyIDs[i];
+                            actor1 = battleStatus->enemyActors[(u8) enemyID1];
+                            priority1 = actor1->turnPriority;
+                            if (actor1 == actor2) {
+                                priority1 += tempPriority;
+                            }
+                            enemyID2 = enemyIDs[j];
+                            actor1 = battleStatus->enemyActors[(u8) enemyID2];
```
Comment (ethteck) [inline src/197F40.c:1997]: Can you declare enemyID2 as a u8? Then you won't need this cast

**Review by ethteck — CHANGES_REQUESTED** [review]
Great job with these! Just a couple suggestions

File: src/197F40.c (hunk 2 above)
Comment (JoshDuMan) [reply src/197F40.c:1997]: Doesn't match unfortunately. Needs the typing it has for later uses.

**Review by ethteck — APPROVED** [review]

**Review by JoshDuMan — APPROVED** [review]

File: src/197F40.c [hunk 3]
```diff
@@ -1942,7 +1942,92 @@ ApiStatus func_8026DA94(Evt* script, s32 isInitialCall) {
...
+    }
+    
+    switch (script->functionTemp[0]) {
+        case 0:
+            script->functionTemp[1] = create_actor(evt_get_variable(script, *args++));
+            script->functionTemp[2] = evt_get_variable(script, *args++);
+            script->functionTemp[0] = 1;
+            break;
+        case 1:
+            actor2 = script->functionTemp[1];
```
Comment (nanaian) [inline src/197F40.c:1968]:
```suggestion
            actor2 = (Actor*) script->functionTemp[1];
```

**Review by nanaian — CHANGES_REQUESTED** [review]
There are a few warnings introduced by these funcs that would be good to fix if you can, e.g.

**Review by nanaian — APPROVED** [review]

**Pull request merged**